### ISSUE TITLE
AD0001 S1144: private record collection throws exception

### ISSUE THREAD

**Issue by OpenEnergyGroup** [SonarSource/sonar-dotnet#8323]
### Description

```
Warning	AD0001	Analyzer 'SonarAnalyzer.Rules.CSharp.UnusedPrivateMember' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.
```

### Repro steps

```
public record Bar(string Value);

public class Foo
{
    private readonly Bar[] Bars = [new("A")];
}
```

### Known workarounds

Seems to be the `private` modifier - works with public. 

### Related information

`<TargetFramework>net8.0</TargetFramework>`

**Comment by zsolt-kolbay-sonarsource** [comment]
Duplicate of #8024
The fix will be released with version 9.13

**Comment by zsolt-kolbay-sonarsource** [comment]
Correction: version 9.13 was released with the fix last week.
